packages/components/src/panel/types.ts

```diff
@@ -0,0 +1,98 @@
+export type PanelProps = {
+	/**
+	 * The text that will be rendered as the title of the panel.
+	 * Text will be rendered inside an `<h2>` tag.
+	 */
+	header?: PanelHeaderProps[ 'label' ];
+	/**
+	 * The CSS class to apply to the wrapper element.
+	 */
+	className?: string;
+	/**
+	 * The content to display within the panel.
+	 */
+	children: React.ReactNode;
+};
+
+export type PanelHeaderProps = {
+	/**
+	 * The text that will be rendered as the title of the panel.
+	 * Will be rendered in an `<h2>` tag.
+	 */
+	label?: string;
+	/**
+	 * The content to display within the panel header.
+	 */
+	children?: React.ReactNode;
+};
+
+export type PanelRowProps = {
+	/**
+	 * The CSS class to apply to the wrapper element.
+	 */
+	className?: string;
+	/**
+	 * The content to display within the panel row.
+	 */
+	children: React.ReactNode;
+};
+
+export type PanelBodyProps = {
+	/**
+	 * Props that are passed to the Button component in the PanelBodyTitle within the panel body.
+	 *
+	 * @default {};
+	 */
+	buttonProps?: object;
```

Comment:
I have this as an object for now, but I believe I'll need to import the types from Button. I'm not sure how to specify it'll need the `BaseButtonProps` and not the `AnchorProps` in the union type `ButtonProps`. From what I understand, it looks like it should check for the tag so not sure why it's being recognized as an anchor. 

But I might be misinterpreting the error related to `Button`:

```
Type '{ children: (string | Element | undefined)[]; onClick: (event: MouseEvent) => void; className: string; slot?: string | undefined; style?: CSSProperties | undefined; ... 255 more ...; ref: ForwardedRef<...>; }' is not assignable to type 'IntrinsicAttributes & (ButtonProps & RefAttributes<any>)'.
  Property 'href' is missing in type '{ children: (string | Element | undefined)[]; onClick: (event: MouseEvent) => void; className: string; slot?: string | undefined; style?: CSSProperties | undefined; ... 255 more ...; ref: ForwardedRef<...>; }' but required in type 'AnchorProps'.
``` 

Perhaps @mirka or @ciampo or @chad1008 could point me in the right direction?

Reply:
Hey! here's a few things that I noticed so far that may help you move forward a little

1. Why are we [typing the `PanelBodyTitle` props](https://github.com/WordPress/gutenberg/blob/44ec4975e1beba42237b7d67d68819426cca3bb6/packages/components/src/panel/body.tsx#L106-L107) using both `PanelBodyTitleProps` and `PanelBodyProps` ? `PanelBodyProps` seems extra. If you intended to use any props from `PanelBodyProps` in `PanelBodyTitle`, you should refactor the types in types.ts and only pick the ones that are needed
2. I noticed that you types [`buttonProps` as `object`](https://github.com/WordPress/gutenberg/blob/44ec4975e1beba42237b7d67d68819426cca3bb6/packages/components/src/panel/types.ts):
  - generally the `object` type is not used in TypeScript. We prefer to use [the `Record` type](https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeys-type)
  - for this specific case, we should probably use [the `ButtonProps` type ](https://github.com/WordPress/gutenberg/blob/c160a7f2602b080b3cd55ee3e72e1825f23c4b42/packages/components/src/button/types.ts#L13)from the `Button` component


I hope this can help you a little bit! Feel free to ask help again to either me or the rest of the team, I'm sure we'll be able to help :)

Reply:
>I'm not sure how to specify it'll need the `BaseButtonProps` and not the `AnchorProps` in the union type ButtonProps. From what I understand, it looks like it should check for the tag so not sure why it's being recognized as an anchor.

Your [ comment here](https://github.com/WordPress/gutenberg/pull/47702/files#r1104990536) answered my initial question quoted above, so I just wanted to share it here to resolve this thread. 🙂 The portion of the comment I am referring to: 

>Knowing that Button can be used to render either a `<button />` or an `<a />`, I think that in this case we're only interested in rendering a button — therefore, I'm thinking of using ButtonAsButtonProps, rather than the whole ButtonProps type

I was initially confused about how to specify importing the button props and not the anchor props, but I see [you added that feature](https://github.com/WordPress/gutenberg/pull/47754) the following week. Awesome!

Reply:
Great, I'm glad I could clarify better how to solve this!